bot/exts/fun/hangman.py

```diff
@@ -0,0 +1,98 @@
+from asyncio import TimeoutError
+from pathlib import Path
+from random import choice
+
+from discord import Color, Embed
+from discord.ext import commands
+
+from bot.bot import Bot
+
+
+class Hangman(commands.Cog):
+    """
+    Cog for the Hangman game.
+
+    Hangman is a classic game where the user tries to guess a word, with a limited amount of tries.
+    """
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        with Path("bot/resources/fun/top_1000_used_words.txt").resolve().open(mode="r", encoding="utf-8") as file:
+            self.all_words = list(map(lambda line: line.strip("\n"), file.readlines()))
+
+    @commands.command()
+    async def hangman(self, ctx: commands.Context, min_length: str = "0", max_length: str = "25",
+                      min_unique_letters: str = "0", max_unique_letters: str = "25") -> None:
+        """
+        Play against the bot, where you have to guess the word it has provided!
+
+        You have 7 tries, and you have to guess the letters that are in the word, or you lose!
+        """
+        # filtering the list of all words depending on the configuration
+        filtered_words = list(filter(lambda x: int(min_length) < len(x) < int(max_length)
+                                     and int(min_unique_letters) < len(set(x)) < int(max_unique_letters),
+                                     self.all_words))
```

Comment:
Once again, this would be way more understandable with a list comprehension and an `if` clause in it, instead of list/filter/lambda combo.